ts/components/ui/BannerErrorState.tsx

```diff
@@ -36,10 +36,15 @@ const styles = StyleSheet.create({
   }
 });
 
+type BannerErrorStateIcons = Extract<
+  IOIcons,
+  "warningFilled" | "noticeFilled" | "errorFilled"
+>;
+
 /* Component Types */
 
 type BaseBannerErrorStateProps = WithTestID<{
-  icon?: IOIcons;
+  icon?: BannerErrorStateIcons;
```

Comment:
Do we really need to limit the icons here? What do you think about making the entire IOIcons set available instead? It would certainly provide more flexibility for potential future use cases.

Reply:
@Hantex9 It certainly provides more flexibility, but the component already has a specific job (`Error...`), which is to signal states related to errors or warnings. For other purposes we already have other components like `Banner` or `Alert`. I can't see `BannerErrorState` being used with a smiley icon, for example.

So I think it's better to limit the choice for now. If we need some flexibility in the future, things will change.

Reply:
It makes sense, and I understand your point about the component having a specific purpose. However, limiting the icons to a predefined set could become a constraint in the future if new error-related icons are introduced or required for additional use cases.

This approach might lead to an increased boilerplate if we need to update the component later to include more icons. Allowing the entire `IOIcons` set would future-proof the component and reduce the need for maintenance in case requirements evolve. What do you think?

Reply:
@Hantex9 Addressed here → [`1ddd447` (#6541)](https://github.com/pagopa/io-app/pull/6541/commits/1ddd4477046f9c82eb50ffaeccce1206af45a541)